### DISCUSSION TITLE
How to use "variables" in MDX / pages for common links, etc?

### DISCUSSION THREAD

**Discussion by viglucci** [facebook/docusaurus#6017] [Q&A] [answered]
Hi all,

I was wondering if there is any support for using "variables" or some other type of placeholder in MDX doc files/pages for things like common URLs, etc? Would be great to avoid having the same URL copy-paste in multiple places, and instead, have a single "source of truth".

Immediately I suppose it is possible to have a `constants.js` file with the URLs exported, and just import them where needed, but I was hoping there might be something a bit more "baked in".

Thanks

**Comment by slorber** [comment, marked as answer]
There are multiple ways you can do that IMHO and it's often requested but I never gave it a try.

You can write a remark plugin that replaces the placeholder of your choice in the code

You can `import * as constants from '@site/src/constants.js'` in your mdx file and read the values as `constants.myUrl`

You can write a remark plugin that adds an import like `import {url1,url2} from '@site/src/constants.js'` at the top of every mdx file and then just access those variables


There are probably other ways to do this, I thought there was a way to provide objects in the MDX scope to be available but I can't find any doc about that so maybe I'm wrong.

**Reply by viglucci** [reply]
Thanks for the response. I think all of these seem sufficiently approachable solutions.
